Added docs Inventory Guide. (#10239)
ansible-collections/community.general: docs/docsite/rst/iocage_inventory_guide_hooks.rst

   zroot/iocage/log                             688K   446G   688K  /zroot/iocage/log
   zroot/iocage/releases                       2.93G   446G   384K  /zroot/iocage/releases
   zroot/iocage/releases/14.2-RELEASE          2.93G   446G   384K  /zroot/iocage/releases/14.2-RELEASE
   zroot/iocage/releases/14.2-RELEASE/root     2.93G   446G  2.88G  /zroot/iocage/releases/14.2-RELEASE/root
   zroot/iocage/templates                       682M   446G   416K  /zroot/iocage/templates
   zroot/iocage/templates/ansible_client        681M   446G   432K  /zroot/iocage/templates/ansible_client
   zroot/iocage/templates/ansible_client/root   681M   446G  3.53G  /zroot/iocage/templates/ansible_client/root

See: `man dhclient-script <https://man.freebsd.org/cgi/man.cgi?dhclient-script>`_

Create the inventory configuration. Use the option :ansopt:`community.general.iocage#inventory:hooks_results` instead of :ansopt:`community.general.iocage#inventory:sudo`:

.. code-block:: console

   shell> cat hosts/02_iocage.yml

.. code-block:: yaml

   plugin: community.general.iocage
   host: 10.1.0.73
   user: admin
   hooks_results:
     - /var/db/dhclient-hook.address.epair0b

> Note:
> 
>    The option hooks_results expects the poolname to be mounted to ``/poolname``. For example, if you
>    activate the pool iocage, this plugin expects to find the hooks_results items in the path
>    /iocage/iocage/jails/<name>/root. If you mount the poolname to a different path, the easiest
>    remedy is to create a symlink.

Reviewer: ```suggestion
.. note::

   The option :ansopt:`community.general.iocage#inventory:hooks_results` expects the poolname to be mounted to ``/poolname``. For example, if you
   activate the pool iocage, this plugin expects to find the :ansopt:`community.general.iocage#inventory:hooks_results` items in the path
   /iocage/iocage/jails/<name>/root. If you mount the poolname to a different path, the easiest
   remedy is to create a symlink.
```

             iocage_ip6: '-'
             iocage_jid: '204'
             iocage_release: 14.2-RELEASE-p3
             iocage_state: up
             iocage_template: ansible_client
             iocage_type: jail
           srv_2:
             iocage_basejail: 'no'
             iocage_boot: 'off'
             iocage_hooks:
             - 10.1.0.204
             iocage_ip4: '-'
             iocage_ip4_dict:
               ip4: []
               msg: DHCP (running -- address requires root)
             iocage_ip6: '-'
             iocage_jid: '205'
             iocage_release: 14.2-RELEASE-p3
             iocage_state: up
             iocage_template: ansible_client
             iocage_type: jail
           srv_3:
             iocage_basejail: 'no'
             iocage_boot: 'off'
             iocage_hooks:
             - 10.1.0.169
             iocage_ip4: '-'
             iocage_ip4_dict:
               ip4: []
               msg: DHCP (running -- address requires root)
             iocage_ip6: '-'
             iocage_jid: '206'
             iocage_release: 14.2-RELEASE-p3
             iocage_state: up
             iocage_template: ansible_client
             iocage_type: jail

> Compose the variable ansible_host
> 
> .. code-block:: console
> 
>    shell> cat hosts/02_iocage.yml

Reviewer: ```suggestion
Compose the variable ``ansible_host``. For example, ``hosts/02_iocage.yml`` could look like:
```

.. code-block:: yaml+jinja

   plugin: community.general.iocage
   host: 10.1.0.73
   user: admin
   hooks_results:
     - /var/db/dhclient-hook.address.epair0b
   compose:
     ansible_host: (iocage_hooks.0 == '-') | ternary(iocage_ip4, iocage_hooks.0)

> Test the jails. Create a playbook:
> 
> .. code-block:: console
> 
>    shell> cat pb-test-uname.yml

Reviewer: ```suggestion
Test the jails. Create a playbook ``pb-test-uname.yml``:
```

..
  Copyright (c) Ansible Project
  GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
  SPDX-License-Identifier: GPL-3.0-or-later

.. _ansible_collections.community.general.docsite.guide_iocage.iocage_inventory_guide.iocage_inventory_guide_hooks:

Hooks
-----

The iocage utility internally opens a console to a jail to get the jail's DHCP address. This
requires root. If you run the command ``iocage list -l`` as unprivileged user, you'll see the message
``DHCP (running -- address requires root)``. If you are not granted the root privilege, use
``/etc/dhclient-exit-hooks``. For example,

.. code-block:: console

   shell> cat /zroot/iocage/jails/srv_1/root/etc/dhclient-exit-hooks

.. code-block:: shell

   case "$reason" in
       "BOUND"|"REBIND"|"REBOOT"|"RENEW")
       echo $new_ip_address > /var/db/dhclient-hook.address.$interface
       ;;
   esac

> where */zroot/iocage* is the activated pool.

Reviewer: I would always use double backticks instead of italics for code.
```suggestion
where ``/zroot/iocage`` is the activated pool.
```
